stream settings: UI changes in stream create/edit form.

NerdyLucifer opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
Issue: #21322
**Testing plan:** <!-- How have you tested? -->

**GIFs or screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
<img width="596" alt="Screenshot 2022-03-07 at 12 50 53" src="https://user-images.githubusercontent.com/85362194/156985886-1db7716a-ce31-4a8a-9a2a-fc9f4aa05458.png">
<img width="306" alt="Screenshot 2022-03-07 at 10 04 47" src="https://user-images.githubusercontent.com/85362194/156985738-9ba00eee-30ca-4d62-bc2f-9d965acf7287.png">



<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

alya commented:
@sahil839 would you be up for reviewing this one?

sahil839 commented on static/templates/stream_settings/stream_types.hbs:
```diff
@@ -39,20 +41,23 @@
-          {{#if disable_message_retention_setting}}disabled{{/if}}>
-            <option value="realm_default">{{#tr}}Use organization level settings {org_level_message_retention_setting}{{/tr}}</option>
+        <select name="stream_message_retention_setting" class="stream_message_retention_setting prop-element">
+            {{#if disable_message_retention_setting}}disabled{{/if}}>
```
There is a typo here. The `disabled` tag is outside the `<select>` tag.

sahil839 commented on static/templates/stream_settings/stream_types.hbs:
```diff
@@ -61,11 +66,9 @@
-              {{#if disable_message_retention_setting}}disabled{{/if}}/>
+            <input type="text" autocomplete="off" name="stream-message-retention-days"
+              class="stream-message-retention-days" value="{{ stream_message_retention_days }}" {{#if
+              disable_message_retention_setting}}disabled{{/if}} />
```
Why you have done these changes? Are you using any other linter that we don't use in Zulip?

NerdyLucifer replied:
I did run the lint test and the tests got passed. I guess the lint test isn't able to identify the differences. BTW i have made some lint fixes in the updates commit.

NerdyLucifer replied:
Got it. I have updated the commits according to what i think would be suitable keeping readability and consistency in my. Please share you feedback.

sahil839 replied:
Thanks. But we should have those lint-only changes (only if we consider it as an improvement) in a separate commit explaining that this has been changed for better readability as I pointed out in above comment as those changes are not related to the others UI improvements that this PR is focussing on.

timabbott replied:
Yeah, I agree, please do a `git rebase -i` and clean up which changes are in which commits, moving all the non-functional changes to a separate commit from the functional ones. This PR is quite hard to read as it is written, even using tools like `git show --color-words`

sahil839 commented:
Also a small suggestion, we generally prefer to have `Fixes` in the last commit only to avoid closing the issue if only some of the commits are merged.

NerdyLucifer commented:
> Also a small suggestion, we generally prefer to have `Fixes` in the last commit only to avoid closing the issue if only some of the commits are merged.

I have removed ```Fixes``` from the 1st two commits.

NerdyLucifer commented:
<img width="316" alt="Screenshot 2022-03-10 at 07 26 29" src="https://user-images.githubusercontent.com/85362194/157573057-9290b11a-36e3-4192-848b-a8024a7da2d7.png">
@timabbott @sahil839 I have updated the commits.
The the text in the dropdowns are at the centre vertically.
Also i have removed the non-functional changes in the code.

timabbott commented:
Merged, with a couple added commits to further clean up some of the subscribers section headings, thanks @NerdyLucifer!

There is a follow-up that I'd appreciate your working on, which is that we actually want the subscribers headings to use this styling from `settings.css`:

```
    h3 {                                                                                                 
        font-size: 1.5em;                                                                                
        font-weight: normal;                                                                             
        line-height: 1.5;                                                                                
    }                                                                                                    
```

But to do that, we'll want to figure out a clean way to extract common CSS to be used for both settings and stream settings in a way that is readable; I'm not sure what the best plan is for that.

NerdyLucifer commented:
@timabbott Can we create a separate css file for common styles which can then be used whenever required.

timabbott commented:
@NerdyLucifer I think the goal would be for `settings.css` to be the file for such common styles among the different settings screens; but maybe it'd be worth creating a `settings_panel.css` file for the common bits; I'm not sure.  Or just putting things in `app_components.css`, which is the global app shared file.
